MaxText/pyconfig.py

```diff
@@ -169,7 +169,8 @@ def user_init(raw_keys):
     '''Transformations between the config data and configs used at runtime'''
 
     # We initialize the jax distributed system here because it must be done before device backend is initialized.
-    if raw_keys["enable_checkpointing"] and raw_keys["async_checkpointing"] and raw_keys["compile_topology_num_slices"]==-1:
+    if (raw_keys["enable_checkpointing"] and raw_keys["async_checkpointing"]
```

Comment:
move this logic into funciton. you can early exit if your criteria rre met.

Reply:
Done, thanks